File: Core/src/main/java/com/alphaz/core/authorization/entity/OauthCodeEntity.java
package com.alphaz.core.authorization.entity;

import javax.persistence.*;
import java.util.Arrays;
import java.util.Objects;

/**
 * @Author: c0der
 * @Date: 下午1:26 2018/2/22
 * @Description:
 */
@Entity
@Table(name = "oauth_code", catalog = "")
public class OauthCodeEntity {
    private String code;
    private byte[] authentication;
    private Long id;

    @javax.persistence.Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }
    @Basic
    @Column(name = "code")
    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    @Basic
    @Column(name = "authentication")
    public byte[] getAuthentication() {
        return authentication;
    }

    public void setAuthentication(byte[] authentication) {
        this.authentication = authentication;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        OauthCodeEntity that = (OauthCodeEntity) o;
        return Objects.equals(code, that.code) &&
                Arrays.equals(authentication, that.authentication);
    }

    @Override
    public int hashCode() {

        int result = Objects.hash(code);
        result = 31 * result + Arrays.hashCode(authentication);
        return result;
    }
}

File: Core/src/main/kotlin/com/alphaz/core/authorization/auth2/OauthCodeEntity.kt
package com.alphaz.core.authorization.auth2

import java.util.*
import javax.persistence.*

/**
 * @Author: c0der
 * @Date: 下午1:26 2018/2/22
 * @Description:
 */
@Entity
@Table(name = "oauth_code", schema = "", catalog = "")
class OauthCodeEntity {
    @get:Basic
    @get:Column(name = "code")
    var code: String? = null
    @get:Basic
    @get:Column(name = "authentication")
    var authentication: ByteArray? = null
    @get:javax.persistence.Id
    @get:GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

}
